### PR TITLE
feat(client-services): logging service

### PR DESCRIPTION
**Pull request by wittjosiah**
### Summary
🔄🆕📝

Added a new feature to query and stream logs from DXOS nodes via gRPC. Updated the `LogLevel` enum values in the `@dxos/log` package to match the `@dxos/protocols` package.

> _`LogLevel` changes_
> _Query and stream logs via gRPC_
> _Autumn of debugging_

### Walkthrough
*  Add gRPC service for querying and streaming logs from DXOS nodes ([link](https://github.com/dxos/dxos/pull/3050/files?diff=unified&w=0#diff-1f246514dbba8145aea2c94eadcc3265b76da0f434106ef87e58db92bdaf5aa0R382-R423))
  - Define `LogLevel` enum to match the one in `@dxos/log` package ([link](https://github.com/dxos/dxos/pull/3050/files?diff=unified&w=0#diff-1f246514dbba8145aea2c94eadcc3265b76da0f434106ef87e58db92bdaf5aa0R382-R423), [link](https://github.com/dxos/dxos/pull/3050/files?diff=unified&w=0#diff-0bc694d37072560e8cdaa1f25144b042858d5c96866cc98a965eb38103058eb0L10-R16))
  - Define `QueryLogsRequest` message to specify log filters ([link](https://github.com/dxos/dxos/pull/3050/files?diff=unified&w=0#diff-1f246514dbba8145aea2c94eadcc3265b76da0f434106ef87e58db92bdaf5aa0R382-R423))
  - Define `LogEntry` message to represent a log message with its metadata ([link](https://github.com/dxos/dxos/pull/3050/files?diff=unified&w=0#diff-1f246514dbba8145aea2c94eadcc3265b76da0f434106ef87e58db92bdaf5aa0R382-R423))
  - Define `LoggingService` service with `QueryLogs` method ([link](https://github.com/dxos/dxos/pull/3050/files?diff=unified&w=0#diff-1f246514dbba8145aea2c94eadcc3265b76da0f434106ef87e58db92bdaf5aa0R382-R423))
* Update `LogLevel` enum values in `@dxos/log` package to be consistent with `@dxos/protocols` package ([link](https://github.com/dxos/dxos/pull/3050/files?diff=unified&w=0#diff-0bc694d37072560e8cdaa1f25144b042858d5c96866cc98a965eb38103058eb0L10-R16))

### CODE REVIEW

**Comment by wittjosiah** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3050** <a href="https://app.graphite.dev/github/pr/dxos/dxos/3050" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dxos/dxos/3050?utm_source=stack-comment).

File: packages/sdk/client-services/src/packlets/logging/logging-service.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import { Event } from '@dxos/async';
+import { Stream } from '@dxos/codec-protobuf';
+import { log, LogEntry as NaturalLogEntry, LogProcessor, shouldLog } from '@dxos/log';
+import { LoggingService, LogEntry, QueryLogsRequest } from '@dxos/protocols/proto/dxos/client/services';
+
+/**
+ * Logging service used to spy on logs of the host.
+ */
+export class LoggingServiceImpl implements LoggingService {
+  private readonly _active = new Set<Event<LogEntry>>();
```
Comment (dmaretskyi) [inline packages/sdk/client-services/src/packlets/logging/logging-service.ts:14]: why not
```suggestion
  private readonly _log = new Event<LogEntry>();
```

and then the stream subscribes to the even

File: packages/sdk/client/src/packlets/client/service-definitions.ts [hunk 2]
```diff
@@ -59,6 +62,7 @@ export const clientServiceBundle = createServiceBundle<ClientServices>({
   SpacesService: schema.getService('dxos.client.services.SpacesService'),
   DataService: schema.getService('dxos.echo.service.DataService'),
   NetworkService: schema.getService('dxos.client.services.NetworkService'),
+  LoggingService: schema.getService('dxos.client.services.LoggingService'),
```
Comment (dmaretskyi) [inline packages/sdk/client/src/packlets/client/service-definitions.ts:65]: I think we can remove the tracing service entirely, and rely on TRACE level logs to cover its functionality

File: packages/sdk/client-services/src/packlets/logging/logging-service.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  constructor() {
+    log.runtimeConfig.processors.push(this._logProcessor);
+  }
+
+  queryLogs(request: QueryLogsRequest): Stream<LogEntry> {
+    const logs = new Event<LogEntry>();
+    this._active.add(logs);
+    return new Stream<LogEntry>(({ ctx, next }) => {
+      const handler = (entry: LogEntry) => {
+        if (shouldLog(transformLogEntry(entry), request.filters)) {
```
Comment (dmaretskyi) [inline packages/sdk/client-services/src/packlets/logging/logging-service.ts:25]: wouldn't it go the other way: you're getting NaturalLogEntry from the logger and converting it to the protobuf LogEntry to stream it

File: packages/sdk/client-services/src/packlets/logging/logging-service.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import { LoggingService, LogEntry, QueryLogsRequest } from '@dxos/protocols/proto/dxos/client/services';
+
+/**
+ * Logging service used to spy on logs of the host.
+ */
+export class LoggingServiceImpl implements LoggingService {
+  private readonly _active = new Set<Event<LogEntry>>();
+
+  constructor() {
+    log.runtimeConfig.processors.push(this._logProcessor);
```
[the remark this answers is not in the document]
Comment (wittjosiah) [reply packages/sdk/client-services/src/packlets/logging/logging-service.ts:17]: So I almost did that, but then I looked at the other services and none of them had open/close methods. When/where would they be called from?

**Pull request merged**